jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/ValidatingCachingContentFactory.java

```diff
@@ -0,0 +1,193 @@
+ * a value of 0 means it is checked on every request,
+ * a positive value indicates the number of milliseconds of the minimum time between validation checks.
+ * </p>
+ * <p>
+ * This also remember a missed entry for the time set by {@code validationTime}ms. After this has
+ * elapsed the entry will be invalid and will be evicted from the cache at the next access.
+ * </p>
+ */
+public class ValidatingCachingContentFactory extends CachingHttpContentFactory implements Runnable
+{
+    private final Scheduler _scheduler;
+    private final long _sweepDelay;
+    private final long _validationTime;
+    private final long _maxCacheIdleTime;
+
+    /**
+     * Construct a {@link ValidatingCachingContentFactory} which validates entries upon use to check if they
+     * are still valid.
+     *
+     * @param authority the wrapped {@link HttpContent.Factory} to use.
+     * @param validationTime time between filesystem checks in ms to see if an {@link HttpContent} is still valid (-1 never validate, 0 always validate).
+     * @param byteBufferPool the {@link org.eclipse.jetty.io.ByteBufferPool} to use.
+     */
+    public ValidatingCachingContentFactory(@Name("authority") HttpContent.Factory authority,
+                                           @Name("validationTime") long validationTime,
+                                           @Name("byteBufferPool") ByteBufferPool byteBufferPool)
+    {
+        this(authority, validationTime, byteBufferPool, null, -1, -1);
+    }
+
+    /**
+     * Construct a {@link ValidatingCachingContentFactory} which validates entries upon use to check if they
+     * are still valid and an optional period sweeper of the cache to find invalid and old entries to evict.
+     *
+     * @param authority the wrapped {@link HttpContent.Factory} to use.
+     * @param validationTime time between filesystem checks in ms to see if an {@link HttpContent} is still valid (-1 never validate, 0 always validate).
+     * @param byteBufferPool the {@link org.eclipse.jetty.io.ByteBufferPool} to use.
+     * @param scheduler scheduler to use for the sweeper, can be null to not use sweeper.
+     * @param sweepDelay time between runs of the sweeper in ms (if 0 never sweep for invalid entries).
+     * @param maxCacheIdleTime amount of time in ms an entry can be unused before evicted by the sweeper (if 0 never evict unused entries).
+     */
+    public ValidatingCachingContentFactory(@Name("authority") HttpContent.Factory authority,
+                                           @Name("validationTime") long validationTime,
+                                           @Name("byteBufferPool") ByteBufferPool byteBufferPool,
+                                           @Name("scheduler") Scheduler scheduler,
+                                           @Name("sweepDelay") long sweepDelay,
+                                           @Name("maxCacheIdleTime") long maxCacheIdleTime)
```

Comment:
Rename to `idleTimeout` like everywhere else.

Reply:
done